workflows/mhcquant.nf

```diff
@@ -57,42 +48,49 @@ if (params.include_proteins_from_vcf)  {
         .set { ch_vcf_from_sheet }
     }
 
-if (params.variant_indel_filter) { variant_indel_filter="-fINDEL" } else { variant_indel_filter="" }
-if (params.variant_frameshift_filter) { variant_frameshift_filter="-fFS" } else { variant_frameshift_filter="" }
-if (params.variant_snp_filter) { variant_snp_filter="-fSNP" } else { variant_snp_filter="" }
+/*
+========================================================================================
+    CONFIG FILES
+========================================================================================
+*/
 
-// Mass Spectronomy data processing options
-x_ions = params.use_x_ions ? '-use_X_ions true' : ''
-z_ions = params.use_z_ions ? '-use_Z_ions true' : ''
-a_ions = params.use_a_ions ? '-use_A_ions true' : ''
-c_ions = params.use_c_ions ? '-use_C_ions true' : ''
-NL_ions = params.use_NL_ions ? '-use_NL_ions true' : ''
-rm_precursor = params.remove_precursor_peak ? '-remove_precursor_peak true' : ''
-fdr_level = (params.fdr_level == 'psm-level-fdrs') ? '' : '-'+params.fdr_level
-fdr_adj_threshold = (params.fdr_threshold == '0.01') ? '0.05' : params.fdr_threshold
+ch_multiqc_config        = file("$projectDir/assets/multiqc_config.yaml", checkIfExists: true)
+ch_multiqc_custom_config = params.multiqc_config ? Channel.fromPath(params.multiqc_config) : Channel.empty()
 
-////////////////////////////////////////////////////
-/* --    IMPORT LOCAL MODULES/SUBWORKFLOWS     -- */
-////////////////////////////////////////////////////
+
+/*
+========================================================================================
+    IMPORT LOCAL MODULES/SUBWORKFLOWS
+========================================================================================
+*/
+// Don't overwrite global params.modules, create a copy instead and use that within the main script.
 def modules = params.modules.clone()
 
+def multiqc_options         = modules['multiqc']
+multiqc_options.args       += params.multiqc_title ? Utils.joinModuleArgs(["--title \"$params.multiqc_title\""]) : ''
+
 def openms_map_aligner_identification_options = modules['openms_map_aligner_identification']
 def openms_comet_adapter_options = modules['openms_comet_adapter']
-
 def generate_proteins_from_vcf_options = modules['generate_proteins_from_vcf']
-
 def percolator_adapter_options = modules['percolator_adapter']
-
 def id_filter_options = modules['id_filter']
 def id_filter_for_alignment_options = id_filter_options.clone()
 def id_filter_whitelist_options = modules['id_filter_whitelist']
 
 id_filter_options.args += " -score:pep " + params.fdr_threshold
-id_filter_for_alignment_options.args += " -score:pep "  + fdr_adj_threshold
+id_filter_for_alignment_options.args += " -score:pep "  + (params.fdr_threshold == '0.01') ? Utils.joinModuleArgs(['-score:pep 0.05']) : Utils.joinModuleArgs(['-score:pep ' + params.fdr_threshold])
+openms_comet_adapter_options.args += params.use_x_ions ? Utils.joinModuleArgs(['-use_X_ions true']) : ''
```

Comment:
I am bit unsure about this `Utils.joinModuleArgs` . The logic should be in the modules.config, but maybe this works better once the new syntax is implemented here.